python/ray/tests/test_usage_stats.py

```diff
@@ -227,6 +239,106 @@ def test_usage_lib_cluster_metadata_generation_usage_disabled(shutdown_only):
     assert len(meta) == 2
 
 
+def test_usage_lib_get_cluster_status(shutdown_only):
+    ray.init(num_cpus=3, num_gpus=1)
+    # Wait for monitor.py to update cluster status
+    time.sleep(10)
+    cluster_status = ray_usage_lib.get_cluster_status(
+        ray.experimental.internal_kv.internal_kv_get_gcs_client(),
+        num_retries=20,
+    )
+    assert cluster_status["num_cpus"] == 3
+    assert cluster_status["num_gpus"] == 1
+    assert cluster_status["total_memory_gb"] > 0
+    assert cluster_status["total_object_store_memory_gb"] > 0
+
+
+def test_usage_lib_get_cluster_config(tmp_path):
+    cluster_config_file_path = tmp_path / "ray_bootstrap_config.yaml"
+    """ Test minimal cluster config"""
+    cluster_config_file_path.write_text(
+        """
+cluster_name: minimal
+max_workers: 1
+provider:
+    type: aws
+    region: us-west-2
+    availability_zone: us-west-2a
+"""
+    )
+    cluster_config = ray_usage_lib.get_cluster_config(cluster_config_file_path)
+    assert cluster_config.get("cloud_provider") == "aws"
+    assert cluster_config.get("min_workers") is None
+    assert cluster_config.get("max_workers") == 1
+    assert cluster_config.get("head_node_instance_type") is None
+    assert cluster_config.get("worker_node_instance_types") is None
+
+    cluster_config_file_path.write_text(
+        """
+cluster_name: full
+min_workers: 1
+provider:
+    type: gcp
+head_node_type: head_node
+available_node_types:
+    head_node:
+        node_config:
+            InstanceType: m5.large
+        min_workers: 0
+        max_workers: 0
+    aws_worker_node:
+        node_config:
+            InstanceType: m3.large
+        min_workers: 0
+        max_workers: 0
+    azure_worker_node:
+        node_config:
+            azure_arm_parameters:
+                vmSize: Standard_D2s_v3
+    gcp_worker_node:
+        node_config:
+            machineType: n1-standard-2
+"""
+    )
+    cluster_config = ray_usage_lib.get_cluster_config(cluster_config_file_path)
```

Comment:
@wuisawesome do we have some shared files to parse? I feel like if any APIs are changed, it will be very difficult to maintain.

Reply:
APIs will change.
It will be difficult to maintain.
Wrapping everything with try excepts was a great idea.

Reply:
I think in the ideal case, autoscaler provides a well defined dataclass for cluster config that everyone else can use. For now I think we can argue that the yaml file format is the public API.